### PR TITLE
Add table gcp_cloud_run_service Closes #479

### LINKED ISSUE #479: Add table gcp_cloudrun_service
**References**
* [Vendor documentation](https://cloud.google.com/run/docs/overview/what-is-cloud-run)
* [Cloud Query example](https://www.cloudquery.io/docs/plugins/sources/gcp/tables/gcp_run_services)
* [Golang SDK](https://pkg.go.dev/google.golang.org/api/run/v1)

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Add passing integration test logs here
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select * from gcp_cloud_run_service
+-------+---------------+----------------+-------------+---------------------------+-------------+-----------------------+----------------------------------------------------------------------------------------------+-------------+------------+---------------------+--->
| name  | client        | client_version | description | create_time               | delete_time | creator               | etag                                                                                         | expire_time | generation | ingress             | se>
+-------+---------------+----------------+-------------+---------------------------+-------------+-----------------------+----------------------------------------------------------------------------------------------+-------------+------------+---------------------+--->
| hello | cloud-console |                |             | 2023-09-04T16:43:16+05:30 | <null>      | madhushree@turbot.com | "CIDb76gGEOiEfg/cHJvamVjdHMvcGFya2VyLWFhYS9sb2NhdGlvbnMvdXMtY2VudHJhbDEvc2VydmljZXMvaGVsbG8" | <null>      | 2          | INGRESS_TRAFFIC_ALL | ht>
+-------+---------------+----------------+-------------+---------------------------+-------------+-----------------------+----------------------------------------------------------------------------------------------+-------------+------------+---------------------+--->

Time: 3.3s. Rows fetched: 1. Hydrate calls: 1.
> select * from gcp_cloud_run_service where name = 'hello' and location = 'us-central1'
+-------+---------------+----------------+-------------+---------------------------+-------------+-----------------------+----------------------------------------------------------------------------------------------+-------------+------------+---------------------+--->
| name  | client        | client_version | description | create_time               | delete_time | creator               | etag                                                                                         | expire_time | generation | ingress             | se>
+-------+---------------+----------------+-------------+---------------------------+-------------+-----------------------+----------------------------------------------------------------------------------------------+-------------+------------+---------------------+--->
| hello | cloud-console |                |             | 2023-09-04T16:43:16+05:30 | <null>      | madhushree@turbot.com | "CIDb76gGEOiEfg/cHJvamVjdHMvcGFya2VyLWFhYS9sb2NhdGlvbnMvdXMtY2VudHJhbDEvc2VydmljZXMvaGVsbG8" | <null>      | 2          | INGRESS_TRAFFIC_ALL | ht>
+-------+---------------+----------------+-------------+---------------------------+-------------+-----------------------+----------------------------------------------------------------------------------------------+-------------+------------+---------------------+--->

```
</details>

### CODE REVIEW

File: gcp/table_gcp_cloudrun_service.go
Comment (madhushreeray30) [inline gcp/table_gcp_cloudrun_service.go:1]: We have missed out on adding some columns - 
- ForceSendFields
- NullFields
- TerminalCondition
- Traffic
- TrafficStatuses

File: gcp/table_gcp_cloudrun_service.go [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+			KeyColumns: plugin.AllColumns([]string{"name", "location"}),
+			Hydrate:    getCloudRunService,
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listCloudRunServices,
+			KeyColumns: plugin.KeyColumnSlice{
+				{
+					Name: "location",
+					Require: plugin.Optional,
+				},
```
Comment (madhushreeray30) [inline gcp/table_gcp_cloudrun_service.go:31]:
```suggestion
				{
					Name:    "location",
					Require: plugin.Optional,
				},
```

File: gcp/table_gcp_cloudrun_service.go [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+			},
+		},
+		GetMatrixItemFunc: BuildComputeLocationList,
+		Columns: []*plugin.Column{
+			{
+				Name:        "name",
+				Description: "The fully qualified name of this Service.",
+				Type:        proto.ColumnType_STRING,
+				Transform: transform.FromField("Name").Transform(lastPathElement),
```
Comment (madhushreeray30) [inline gcp/table_gcp_cloudrun_service.go:40]:
```suggestion
				Transform:   transform.FromField("Name").Transform(lastPathElement),
```

File: gcp/table_gcp_cloudrun_service.go [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+	matrixLocation := d.EqualsQualString(matrixKeyLocation)
+	// Since, when the service API is disabled, matrixLocation value will be nil
+	if matrixLocation != "" {
+		location = matrixLocation
+	}
+
+	// Minimize API call as per given location
+	if region != "" && region != location{
+		return nil, nil
+	}
```
Comment (madhushreeray30) [inline gcp/table_gcp_cloudrun_service.go:248]:
```suggestion
	if region != "" && region != location {
		return nil, nil
	}
```

File: docs/tables/gcp_cloud_run_service.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+```sql
+select
+  name,
+  t ->> 'Percent' as percent,
+  t ->> 'Revision' as revision,
+  t ->> 'Tag' as tag,
+  t ->> 'Type' as type
+from
+  gcp_cloud_run_service,
+  jsonb_array_elements(traffic) as t;
```
Comment (madhushreeray30) [inline docs/tables/gcp_cloud_run_service.md:134]: Column traffic needs to be added

**Review by madhushreeray30 — COMMENTED** [review]
@ParthaI Please have a look at the comments, thanks!

**Review by misraved — COMMENTED** [review]
@ParthaI please take a look at the review comments. Thanks!!

File: gcp/table_gcp_cloudrun_service.go
Comment (ParthaI) [reply gcp/table_gcp_cloudrun_service.go:1]: @madhushreeray30 , I had added `traffic_statuses` column earlier, rest of missing column I added in.

File: gcp/service.go [hunk 5]
```diff
@@ -23,7 +24,7 @@ import (
 	"google.golang.org/api/logging/v2"
 	"google.golang.org/api/monitoring/v3"
 	"google.golang.org/api/pubsub/v1"
-	"cloud.google.com/go/redis/apiv1"
+	"google.golang.org/api/run/v2"
```
[the remark this answers is not in the document]
Comment (ParthaI) [reply gcp/service.go:27]: It has been shifted to top automatically after prettifying the code. Nothing is breaking the package name is correct.

**Review by madhushreeray30 — APPROVED** [review]

**Pull request merged**